### ISSUE TITLE
DP-ed Relations in components in non-DP types show up `null` in REST API

### ISSUE THREAD

**Issue by laurenskling** [strapi/strapi#21382]
### Required System information

- Node.js version: 20
- NPM version: yarn 1.22
- Strapi version: 5.0.0 (and rc 17-29 at least)
- Database: sqlite
- Operating system: macos
- Is your project Javascript or Typescript: js

### Describe the bug + Steps to reproduce the behavior

follow up bug report to https://github.com/strapi/strapi/issues/21355

I have a Homepage singleType. It doesnt have draftAndPublish, because the homepage never goes unpublished (v4 logic of DP).

 have a repeatable component with relations inside. (see https://github.com/strapi/strapi/issues/21354)
After migration, when I request the REST API all return correctly. (on `http://0.0.0.0:1337/api/homepage/homepage?populate[sections][populate][section]=*`). When I change one of the relations in the UI, that relation goes `null` in the REST API result. So after changing all the relations in the UI, all of them will return f.e. `{ "id": 2, "section": null },`. To make things confusing, when I add `&status=draft` to the REST call, the _last saved relation_ is returned. So if I change component 2 to another relation, with `&status=draft`, it will now return the relation, but component 2 will be `null` again when i change component 3.

Make it make sense.

### Expected behavior

A non-DP contentType should always return all of the requested relations.

### Aditional context

will create a reproduction codebase a.s.a.p.

**Comment by laurenskling** [comment]
I'm pretty sure, as https://github.com/strapi/strapi/issues/21355 looks like a unidirectional bug, this is the same problem. As relations in components are always unidirectional.

**Comment by laurenskling** [comment]
okay I did some more testing and this really does seem to be related to non-DP. I've tried collectionType/singleType, no difference. But when I change `draftAndPublish` from true to false, relations are showing up as `null`

**Comment by laurenskling** [comment]
Found the combination!
Repo: https://github.com/laurenskling/strapi-5-tests

The exact situation is (based on the content types in this repo):
1. have DP-ed Authors (non-DP authors work)
2. have a non-DP Articles (DP will work)
3. have a (repeatable) component to the author(s) in articles (components in DZ's _do_ work! also in the repo)
4. add 2 authors
5. add an article with 2 quotes, relate them to the authors

<img width="844" alt="Scherm­afbeelding 2024-09-19 om 21 39 54" src="https://github.com/user-attachments/assets/b2a4b10c-f868-4add-962f-1fc2205cb80b">

```
// http://0.0.0.0:1337/api/articles?populate[quotes][populate][author][fields]=*&populate[blocks][on][shared.quote][populate][author]=*

{
  "data": [
    {
      "id": 24,
      "documentId": "wjfpxia0jujnrhwlouv3in1o",
      "title": "quote me!",
      "description": null,
      "slug": null,
      "createdAt": "2024-09-19T19:30:05.362Z",
      "updatedAt": "2024-09-19T19:39:47.118Z",
      "publishedAt": "2024-09-19T19:39:47.105Z",
      "locale": null,
      "quotes": [
        {
          "id": 18,
          "title": "this is a direct component",
          "body": "by john",
          "author": null
        },
        {
          "id": 21,
          "title": "this is a direct component 2",
          "body": "by jane",
          "author": null
        }
      ],
      "blocks": [
        {
          "__component": "shared.quote",
          "id": 20,
          "title": "This is a DZ quote",
          "body": "by jane",
          "author": {
            "id": 11,
            "documentId": "iwxdy1ep3h64th8j1pyvbaxr",
            "name": "Jane",
            "email": null,
            "createdAt": "2024-09-19T19:29:37.798Z",
            "updatedAt": "2024-09-19T19:29:37.798Z",
            "publishedAt": null,
            "locale": null
          }
        }
      ]
    }
  ],
  "meta": {
    "pagination": {
      "page": 1,
      "pageSize": 25,
      "pageCount": 1,
      "total": 1
    }
  }
}
```

**Comment by laurenskling** [comment]
And like I mentioned before, when requesting draft, the relations will show up:

```
// http://0.0.0.0:1337/api/articles?populate[quotes][populate][author][fields]=*&populate[blocks][on][shared.quote][populate][author]=*&status=draft

{
  "data": [
    {
      "id": 24,
      "documentId": "wjfpxia0jujnrhwlouv3in1o",
      "title": "quote me!",
      "description": null,
      "slug": null,
      "createdAt": "2024-09-19T19:30:05.362Z",
      "updatedAt": "2024-09-19T19:39:47.118Z",
      "publishedAt": "2024-09-19T19:39:47.105Z",
      "locale": null,
      "quotes": [
        {
          "id": 18,
          "title": "this is a direct component",
          "body": "by john",
          "author": {
            "id": 10,
            "documentId": "zpjqh2wgj7fl70iycoh7hjk5",
            "name": "John",
            "email": null,
            "createdAt": "2024-09-19T19:29:33.053Z",
            "updatedAt": "2024-09-19T19:29:33.053Z",
            "publishedAt": null,
            "locale": null
          }
        },
        {
          "id": 21,
          "title": "this is a direct component 2",
          "body": "by jane",
          "author": {
            "id": 11,
            "documentId": "iwxdy1ep3h64th8j1pyvbaxr",
            "name": "Jane",
            "email": null,
            "createdAt": "2024-09-19T19:29:37.798Z",
            "updatedAt": "2024-09-19T19:29:37.798Z",
            "publishedAt": null,
            "locale": null
          }
        }
      ],
      "blocks": [
        {
          "__component": "shared.quote",
          "id": 20,
          "title": "This is a DZ quote",
          "body": "by jane",
          "author": {
            "id": 11,
            "documentId": "iwxdy1ep3h64th8j1pyvbaxr",
            "name": "Jane",
            "email": null,
            "createdAt": "2024-09-19T19:29:37.798Z",
            "updatedAt": "2024-09-19T19:29:37.798Z",
            "publishedAt": null,
            "locale": null
          }
        }
      ]
    }
  ],
  "meta": {
    "pagination": {
      "page": 1,
      "pageSize": 25,
      "pageCount": 1,
      "total": 1
    }
  }
}
```

**Comment by laurenskling** [comment]
https://github.com/strapi/strapi/issues/21355#issuecomment-2374353338 <- working experimental

**Comment by laurenskling** [comment]
Seems to be fixed since 5.0.3
